server/tpm/js_ek_tpm_windows.go

```diff
@@ -0,0 +1,283 @@
+
+// Gets/Regenerates the Storage Root Key (SRK) from the TPM. Caller MUST flush this handle when done.
+func regenerateSRK(rwc io.ReadWriteCloser, srkPassword string) (tpmutil.Handle, error) {
+	// Default EK template defined in:
+	// https://trustedcomputinggroup.org/wp-content/uploads/Credential_Profile_EK_V2.0_R14_published.pdf
+	// Shared SRK template based off of EK template and specified in:
+	// https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-v2.0-Provisioning-Guidance-Published-v1r1.pdf
+	srkTemplate := tpm2.Public{
+		Type:       tpm2.AlgRSA,
+		NameAlg:    tpm2.AlgSHA256,
+		Attributes: tpm2.FlagFixedTPM | tpm2.FlagFixedParent | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth | tpm2.FlagRestricted | tpm2.FlagDecrypt | tpm2.FlagNoDA,
+		AuthPolicy: nil,
+		// We must use RSA 2048 for the intel TSS2 stack
+		RSAParameters: &tpm2.RSAParams{
+			Symmetric: &tpm2.SymScheme{
+				Alg:     tpm2.AlgAES,
+				KeyBits: 128,
+				Mode:    tpm2.AlgCFB,
+			},
+			KeyBits:    2048,
+			ModulusRaw: make([]byte, 256),
+		},
+	}
+	// Create the parent key against which to seal the data
+	srkHandle, _, err := tpm2.CreatePrimary(rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", srkPassword, srkTemplate)
+	return srkHandle, err
+}
+
+type natsTPMPersistedKeys struct {
+	Version    int    // json: "version"
+	PrivateKey []byte // json: "privatekey"
+	PublicKey  []byte // json: "publickey"
+}
+
+// Writes the private and public blobs to disk in a single file. If the directory does
+// not exist, it will be created. If the file already exists it will be overwritten.
+func writeTPMKeysToFile(filename string, privateBlob []byte, publicBlob []byte) error {
+	keyDir := filepath.Dir(filename)
+	if err := os.MkdirAll(keyDir, 0755); err != nil {
+		return fmt.Errorf("unable to create/access directory %q: %v", keyDir, err)
+	}
+
+	// Create a new set of persisted keys. Note that the private key doesn't necessary
+	// need to be protected as the TPM password is required to use unseal, although it's
+	// a good idea to put this in a secure location accessible to the server.
+	tpmKeys := natsTPMPersistedKeys{
+		Version:    JsKeyTPMVersion,
+		PrivateKey: []byte(base64.StdEncoding.EncodeToString(privateBlob)),
+		PublicKey:  []byte(base64.StdEncoding.EncodeToString(publicBlob)),
+	}
+	// Convert to JSON
+	keysJSON, err := json.Marshal(tpmKeys)
+	if err != nil {
+		return fmt.Errorf("unable to marshal keys to JSON: %v", err)
+	}
+	// Write the JSON to a file
+	if err := os.WriteFile(filename, keysJSON, 0644); err != nil {
+		return fmt.Errorf("unable to write keys file to %q: %v", filename, err)
+	}
+	return nil
+}
+
+// Reads the private and public blobs from a single file. If the file does not exist,
+// or the file cannot be read and the keys decoded, an error is returned.
+func readTPMKeysFromFile(filename string) ([]byte, []byte, error) {
+	if _, err := os.Stat(filename); os.IsNotExist(err) {

```

Comment:
Wondering if this is necessary or whether `os.ReadFile(...)` would return an `os.IsNotExist(err)` too.